### PR TITLE
Fix upsample kernel launch / reorder arguments

### PR DESCRIPTION
**Pull request by xmnlab**
this is a follow up for https://github.com/pytorch/pytorch/pull/19630

### CODE REVIEW

**Comment by xmnlab** [comment]
it seems `ci/circleci: pytorch_linux_xenial_cuda9_2_cudnn7_py3_gcc7_build ` passed ... but it didn't send this info back for this error: `GitHub rate limited us while adding a commit status of success: You have triggered an abuse detection mechanism. Please wait a few minutes before you try again.`

I will rerun the tests to see if it is working now.

@pytorchbot retest this please

**Comment by xmnlab** [comment]
it is raising the same error: `
GitHub rate limited us while adding a commit status of success: You have triggered an abuse detection mechanism. Please wait a few minutes before you try again.`

**Comment by xmnlab** [comment]
@skrah could you review this PR?

File: aten/src/ATen/native/cuda/UpSample.cuh [hunk 1]
```diff
@@ -158,51 +158,42 @@ __device__ __forceinline__ static int nearest_neighbor_compute_source_index(
...
-    int y) {
-  int access_x =
-      max<int>(min<int>(x, width - 1), static_cast<int>(0));
-  int access_y =
-      max<int>(min<int>(y, height - 1), static_cast<int>(0));
+    int width,
+    int y,
+    int x) {
+  int access_x = max<int>(min<int>(x, width - 1), static_cast<int>(0));
+  int access_y = max<int>(min<int>(y, height - 1), static_cast<int>(0));
```
Comment (skrah) [inline aten/src/ATen/native/cuda/UpSample.cuh]: If you change the argument order to match the indexing order in `data[][][][]`, I'd also put `int access_y ...` above `int access_x ...`.  That makes it absolutely clear that the order is intentional.

And please remove the cast in `static_cast<int>(0)`.  An integer literal without a suffix is already an `int` if it fits the integer size.

File: aten/src/ATen/native/cuda/UpSample.cuh [hunk 2]
```diff
@@ -158,51 +158,42 @@ __device__ __forceinline__ static int nearest_neighbor_compute_source_index(
...
     accscalar_t value) {
-  int access_x =
-      max<int>(min<int>(x, width - 1), static_cast<int>(0));
-  int access_y =
-      max<int>(min<int>(y, height - 1), static_cast<int>(0));
-  /* TODO: result here is trucated to scalar_t, 
-     check: https://github.com/pytorch/pytorch/pull/19630#discussion_r281426912 */
+  int access_x = max<int>(min<int>(x, width - 1), static_cast<int>(0));
+  int access_y = max<int>(min<int>(y, height - 1), static_cast<int>(0));
+  /* TODO: result here is trucated to scalar_t,
```
Comment (skrah) [inline aten/src/ATen/native/cuda/UpSample.cuh]: `truncated`

**Comment by skrah** [comment]
@xmnlab  I have two minor nitpicks, LGTM otherwise.

File: aten/src/ATen/native/cuda/UpSample.cuh (hunk 1 above)
Comment (xmnlab) [reply aten/src/ATen/native/cuda/UpSample.cuh]: you're right. I will be better change the order here too. thanks!
sure! I will remove it .. probably it was done by my script ... also I will check the script and remove this rule. thanks!

File: aten/src/ATen/native/cuda/UpSample.cuh (hunk 2 above)
Comment (xmnlab) [reply aten/src/ATen/native/cuda/UpSample.cuh]: do you have any suggestion related to remove the truncation on each addition?
if we need to rewrite the kernel, maybe would be good to open an issue for that and treat that in another PR.

**Comment by xmnlab** [comment]
thanks @skrah for the review!

File: aten/src/ATen/native/cuda/UpSample.cuh (hunk 1 above)
Comment (xmnlab) [reply aten/src/ATen/native/cuda/UpSample.cuh]: I recall now, before it was `static_cast<int_64>(0)` and I changed all index from `int64_t` to `int` .. and I forgot to fix these cast. thanks!

File: aten/src/ATen/native/cuda/UpSample.cuh (hunk 2 above)
Comment (xmnlab) [reply aten/src/ATen/native/cuda/UpSample.cuh]: I will open a new issue for that this truncate item

**Comment by skrah** [comment]
One more thing. Looking at the porting issue again, I think that Edward suggested that all launch sites should be fixed in the same manner.  See e.g. here:

https://github.com/pytorch/pytorch/pull/19630/files#r281619993

So `C10_LAUNCH_BOUNDS_1(1024)` should be unconditional in all files you ported and not only the ones that had test failures.

**Comment by jjsjann123** [comment]
> So C10_LAUNCH_BOUNDS_1(1024) should be unconditional in all files you ported and not only the ones that had test failures.

I'm getting error on bilinear2d in test_nn.py where launch_bound is needed for CUDA as well. I'm running on volta.
We should remove the HIP guard.

**Comment by xmnlab** [comment]
thanks @skrah for caching that. thanks @jjsjann123 for the feedback. I am fixing that now.

**Comment by skrah** [comment]
Thanks, LGTM.  Let's ask @ezyang to continue.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**